### ISSUE TITLE
`sls plugin search` fails if a plugin is uninstalled

### ISSUE THREAD

**Issue by ryanmurakami** [serverless/serverless#4259]
# This is a Bug Report

## Description

In a given directory that has a plugin listed in the `serverless.yml` which is not installed, a call to `sls plugin search --query webpack` fails with the message:

`Serverless plugin "serverless-spotinst-functions" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file.`

I would expect this plugin search to complete even if I have an uninstalled plugin locally.

## Additional Data

* ***Serverless Framework Version you're using***: 1.22
* ***Node.js Version you're using***: 8.5.0
* ***Operating System***: macos

**Comment by pmuens** [comment]
Thanks for reporting @ryanmurakami 👍

Nice catch! /cc @horike37

**Comment by horike37** [comment]
All plugin commands(list, search, install, uninstall) might want not to catch the error in the above situation. Otherwise, users who set plugins manually within the yaml file would face the same error.
What do you think?

**Comment by ryanmurakami** [comment]
@horike37 I agree. Just to add context, this happened with a new service from a template that had a plugin hardcoded in the `serverless.yml`. So, this could happen to someone who didn't even know there was an uninstalled plugin.

**Comment by pmuens** [comment]
Good idea @horike37 👍

Just wondering if all commands should skip the check if you're in a service. I'd say that `install` and `uninstall` should still check if the command was run in a service directory (what should happen if I run them globally 🤔?).

But `list` and `search` should should definitely work globally 👍

**Comment by ryanmurakami** [comment]
@pmuens I like the idea of `install` and `uninstall` checking the location of execution, just not the state of other plugins. I think the plugin commands shouldn't fail based on the state of another plugin.

**Comment by horike37** [comment]
### Conclusion for the implementation 
- `install` and `uninstall` should check the location of execution (if you are in services)
- All plugin commands shouldn't catch the error related to the state of another plugin.

**Comment by psyhomb** [comment]
Is there anything new on this issue?

**Comment by tmclaugh** [comment]
Just ran into a related issue.  I added some plugins I know I need to install while working on my project's serverless.yml.  I had to comment the section out in order to run `sls plugin install` to get the plugins I needed and then uncomment.  I'm a little unclear on what people think the default behavior of `install` should be.  I think it should skip the plugin check so people can install plugins easily.  Or, is there another command that I can run that'll look at what should be installed and install the plugins?
